### DISCUSSION TITLE
ScriptReference/Unity.Collections.Allocator.Temp

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#7] [Notes]
# ScriptReference/Unity.Collections.Allocator.Temp

https://docs.unity3d.com/ScriptReference/Unity.Collections.Allocator.Temp.html

**Comment by vertxxyz** [comment]
**`Allocator.Temp`** has the fastest allocation. Use it for allocations with a lifespan of one frame or fewer. However, you can’t use `Temp` to pass `NativeContainer` allocations to jobs.
(See [NativeContainer](https://docs.unity3d.com/Manual/JobSystemNativeContainer.html).)

**Comment by vertxxyz** [comment]
From **kevinmv** on the Unity Discord:
`Allocator.Temp` is indeed thread local. 
The allocation is not stack based, as of 2022.1 it uses a bump allocator that is reset based on the current scope. Most temp allocations are reset on the next player update. For jobs, you can allocate using temp inside your `Execute` as we provide a new temp scope for allocations that live only for the scope of the `Execute` method itself. And indeed, since temp allocations are thread local, we fail scheduling any native containers [passed into a job] that use `Allocator.Temp` as you are almost certainly guaranteed to have problems if you refer to temp memory from another thread.
Using `Allocator.Temp` inside `Execute` is encouraged if you really need to allocate (best to avoid allocations inside a job when possible but I understand that isn't always the case). The temp allocation will be quite quick since it usually amounts to a pointer addition rather than fetching a new block of memory from the heap.
And since it is thread local, there is no contention on such an allocation unlike `TempJob` which can be quite slow but provides different lifetime guarantees (4 frames for the alloc before we tell you you likely forgot to free your allocation).
